core/resource.h

```diff
@@ -108,6 +108,7 @@ class Resource : public Reference {
 
 	void set_local_to_scene(bool p_enable);
 	bool is_local_to_scene() const;
+	bool is_resource() const override { return true; }
```

Comment:
This is unnecessary, you can use `Object::cast_to<Resource>(obj)` to know if an object inherits Resource.
See https://docs.godotengine.org/en/latest/community/contributing/best_practices_for_engine_contributors.html#solutions-must-be-local